src/linalg/pow.rs

```diff
@@ -0,0 +1,72 @@
+//! This module provides the matrix exponential (pow) function to square matrices.
+
+use std::ops::DivAssign;
+
+use crate::{allocator::Allocator, DefaultAllocator, DimMin, MatrixN};
+use num::PrimInt;
+use simba::scalar::ComplexField;
+
+impl<N: ComplexField, D> MatrixN<N, D>
+where
+    D: DimMin<D, Output = D>,
+    DefaultAllocator: Allocator<N, D, D>,
+    DefaultAllocator: Allocator<N, D>,
+{
+    /// Attempts to raise this matrix to an integral power `e` in-place. If this
+    /// matrix is non-invertible and `e` is negative, it leaves this matrix
+    /// untouched and returns `false`. Otherwise, it returns `true` and
+    /// overwrites this matrix with the result.
+    pub fn pow_mut<T: PrimInt + DivAssign>(&mut self, mut e: T) -> bool {
```

Comment:
Instead of returning a `bool`, it might be better to return a `Result<(), ()>` or `Option<()>` for improved semantics (`unwrap` etc.) and the `must_use` annotation.

Reply:
I've implemented this suggestion. `pow_mut` now returns a `Result<(), ()>`.